### PR TITLE
Scaffolding #680

### PR DESCRIPTION
**Pull request by Stolas**
Basic rough version of scaffolding.

Missing:

- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: pwnlib/scaffolds/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python2
+import sys
+
+# Todo: Prettify
+def question(line, default):
```
Comment (zachriggle) [inline pwnlib/scaffolds/__init__.py:5]: This is already handled by the `ui` module, e.g. `ui.options` and `ui.yesno`

File: pwnlib/scaffolds/remote.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python
+
+class RemoteExploitTemplate():
```
Comment (zachriggle) [inline pwnlib/scaffolds/remote.py:3]: I see no point in declaring new classes, especially without any base / interface type.

Also, I don't see anything provided here which is not handled better and in a more standardized fashion with Mako templates.

**Comment by Stolas** [comment]
I see your point. But don't you think that:


```
<%
    host = question("Hostname", "example.com")
    port = question("Port", 80)
%>
#!/usr/bin/env python

from . import question

class WebExploitTemplate():
    name = 'web'
    summary = 'New HTTP-based exploit'

    def build(self):
        base = """
#!/usr/bin/env python
from pwn import *

http_pkt = "GET / HTTP/1.1\\r\\n"\\
           "User Agent: %s\\r\\n" % useragents.random()

sock = remote("${host}", ${port})
sock.send(http_pkt)
log.info(sock.recvline())
```

Looks a tad odd?
Also the classes are required for the way they are found and handled, they require a base class indeed.

**Comment by zachriggle** [comment]
There's no need for base classes, since almost none of them share anything in common (host name:port tuple, versus process and arguments, vs. username-host-password).  However, if you're taking the class approach, there should clearly be a defined interface, rather than duck typing them all.

Regarding the template, I think your example is significantly more complex than would be required.  Consider:

```mako
<%page args="host, port"/>
from pwn import *

http_pkt = "GET / HTTP/1.1\r\n"
http_pkt += "User Agent: %s\r\n" % useragents.random()

sock = remote("${host}", ${port})
sock.send(http_pkt)
log.info(sock.recvline())
```

In any case, most of the utility of `PwnUp` as far as I can tell lies in one of two features:

1. Emitting a dumb template with almost nothing in it (as above)
2. Emitting a template with `.send` and `.recvuntil` where the contents are determined dynamically via an interactive session

**Pull request closed without merging**